### PR TITLE
Add interpolation page to glossary

### LINKED ISSUE #15208: Add "Interpolation" page to the the glossary
## What is the new suggestion?
I suggest adding a new page to the glossary about interpolation

## Why is it important or useful?
This issue is related to https://github.com/mdn/content/issues/15168. Also I've explored the existing mdn pages and found out that there are a places where a link to the page about interpolation would be helpful. For example in [calcMode](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/calcMode) or [CSS filters](https://developer.mozilla.org/en-US/docs/Web/CSS/filter#interpolation). I think this term is somewhat opaque without a context. Moreover I believe it is totally mystical for newcomers and those who doesn't have a good knowledge of math.

MDN URL: https://developer.mozilla.org/en-US/docs/Glossary

### PR DESCRIPTION
**Pull request by myshov**
#### Summary
Add interpolation page to glossary

#### Motivation
I suggest adding a new page to the glossary about interpolation because on MDN there are places where the term "Interpolation" is used without a context. A link to glossary page in this cases would be helpful.

#### Related issues
Fixes #15208 

#### Metadata
- [x] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

**Comment by myshov** [comment]
It seems that the flaws are not related to the current changes

**Comment by wbamberg** [comment]
> It seems that the flaws are not related to the current changes

Yes, you don't have to worry about these.

@fiji-flo , @Rumyra , speaking of flaws: people often have this question when they file PRs. I think the flaw system is quite contributor-hostile in this way.

**Comment by myshov** [comment]
@Rumyra hey! 

> with two sub pages - one that describes value interpolation, and one for string interpolation

I am not sure I understand this correctly. Do you mean subpages with this directory structure?

```
interpolation
 |
 |--value_interpolation
 |    |
 |    |-index.md
 |
 |--string_interpolation
      |
      |-index.md
```

> Would you be able to modify the pr to do so?

Yes, no problem :)

**Comment by Rumyra** [comment]
Hi @myshov :) It would be:

```
glossary
  |
  |--interpolation
       |
       |--index.md
       |
       |--value_interpolation
       |    |
       |    |-index.md
       |
       |--string_interpolation
            |
            |-index.md
```

A good example is the signature definition: https://developer.mozilla.org/en-US/docs/Glossary/Signature 👍

File: files/en-us/glossary/interpolation/index.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+title: Interpolation
+slug: Glossary/Interpolation
+tags:
+  - Glossary
+  - Animation
+  - Gradient
+  - JavaScript
+---
+Interpolation is a process of calculating values based on known values. The interpolation is used to obtain intermediate values of height, width etc. during the process of their animation. In the gradients interpolation is used to obtain intermediate values of colors based on a provided list of colors. Also the term "interpolation" is used as a description of substitution of the strings in the template literals.
```
Comment (estelle) [inline files/en-us/glossary/interpolation/index.md:10]:
```suggestion
Interpolation is a process of calculating values based on known values. The interpolation is used to obtain intermediate values of height, width, etc. during the process of their animation. In gradients, interpolation is used to obtain intermediate values of colors based on a provided list of colors. Also, the term "interpolation" is used as a description of the substitution of strings in the template literals.
```

**Review by estelle — APPROVED** [review]

**Comment by estelle** [comment]
Thank you!

**Comment by estelle** [comment]
I didn't see this discussion until after merging. Reversion is at https://github.com/mdn/content/pull/15446

**Pull request merged**